app/models/log.rb

```diff
@@ -125,9 +126,11 @@ def address_string
   end
 
   def address_options
-    return @address_options if @address_options
+    return @address_options if @address_options && @last_searched_address_string == address_string
```

Comment:
Sometimes the address input fields change and we don't get the new address options in time. So previously this situation would happen:
1. Search for address that returns some options
2. Land on upon-selection page
3. Go back to address search page and enter a query that wouldn't return any options
4. Next page is no-address-found, but because we still have old @address_options set, this page would not get routed to, so the flow would exit into the tasklist page